Library/Homebrew/cask/audit.rb

```diff
@@ -282,6 +284,66 @@ def check_token_conflicts
       add_warning "possible duplicate, cask token conflicts with Homebrew core formula: #{core_formula_url}"
     end
 
+    def check_token_valid
+      return unless @strict
+
+      add_warning "cask token is not lowercase" if cask.token.downcase!
+
+      add_warning "cask token contains non-ascii characters" unless cask.token.ascii_only?
+
+      add_warning "cask token + should be replaced by -plus-" if cask.token.include? "+"
+
+      add_warning "cask token @ should be replaced by -at-" if cask.token.include? "@"
+
+      add_warning "cask token whitespace should be replaced by hyphens" if cask.token.include? " "
+
+      add_warning "cask token underscores should be replaced by hyphens" if cask.token.include? "_"
+
+      if cask.token.match?(/[^a-z0-9\-]/)
+        add_warning "cask token should only contain alphanumeric characters and hyphens"
+      end
+
+      add_warning "cask token should not contain double hyphens" if cask.token.include? "--"
+
+      return unless cask.token.end_with?("-") || cask.token.start_with?("-")
+
+      add_warning "cask token should not have leading or trailing hyphens"
+    end
+
+    def check_token_bad_words
+      return unless @strict
+
+      add_warning "cask token contains .app" if cask.token.end_with? ".app"
+
+      if cask.token.end_with? "alpha", "beta", "release candidate"
+        add_warning "cask token contains version designation"
+      end
+
+      add_warning "cask token mentions launcher" if cask.token.end_with? "launcher"
+
+      add_warning "cask token mentions desktop" if cask.token.end_with? "desktop"
+
+      add_warning "cask token mentions platform" if cask.token.end_with? "mac", "osx"
+
+      add_warning "cask token mentions architecture" if cask.token.end_with? "x86", "32-bit", "ppc"
+
+      if cask.token.end_with?("cocoa", "qt", "gtk", "wx", "java") && !%w[cocoa qt gtk wx java].include?(cask.token)
+        add_warning "cask token mentions framework"
+      end
+
+      return unless @cask.tap
+
+      tokens = @cask.tap.cask_tokens
+      clean_tokens = []
+      tokens.each do |s|
+        clean_tokens << s.dup.delete("-")
+      end
+
+      return unless clean_tokens.include? "#{@cask.tap.name}/#{cask.token.dup.delete("-")}"
+
+      add_warning "cask token differs from existing only by hyphenation"
```

Comment:
I think we can remove this one. I’m not confident we follow this one to the letter and I’m OK with removing it from the reference.

Reply:
Wouldn't this be a good opportunity to start enforcing it then?

Reply:
If it hasn’t been enforced and hasn’t caused issues, why start now?

If we retroactively apply it, we’ll break casks (we don’t yet have a `formula_renames` equivalent); if we don’t, we might get issues complaining that so-and-so doesn’t conform to it.

It would effectively be a new rule. One that exists to prevent a situation that hasn’t proven to be problematic. Also, by removing you can get rid of some code here.

https://github.com/Homebrew/homebrew-cask/pull/83966